module/zfs/zfs_znode.c

```diff
@@ -526,26 +525,12 @@ zfs_inode_update_impl(znode_t *zp, boolean_t new)
 	if (zfsctl_is_node(ip))
 		return;
 
-	sa_lookup(zp->z_sa_hdl, SA_ZPL_ATIME(zsb), &atime, 16);
-	sa_lookup(zp->z_sa_hdl, SA_ZPL_MTIME(zsb), &mtime, 16);
-	sa_lookup(zp->z_sa_hdl, SA_ZPL_CTIME(zsb), &ctime, 16);
-
 	dmu_object_size_from_db(sa_get_db(zp->z_sa_hdl), &blksize, &i_blocks);
 
 	spin_lock(&ip->i_lock);
 	ip->i_mode = zp->z_mode;
 	zfs_set_inode_flags(zp, ip);
 	ip->i_blocks = i_blocks;
-
-	/*
-	 * Only read atime from SA if we are newly created inode (or rezget),
-	 * otherwise i_atime might be dirty.
-	 */
-	if (new)
```

Comment:
Yeah, Intended to do this change and add it do this PR but I wanted to clear the other commits first. So, the reason why I locked the zfs_write is that the inode->i_mtime etc attributes are being changed. And just to be on the safe side I wanted to lock the vfs inode while the operations is in progress, otherwise it might be possible to initiate 2 separate operations which might lead to corruption. E.g. a write + truncate (maybe, I'm only speculating)